relengapi/blueprints/slaveloan/tasks.py

```diff
@@ -83,6 +83,33 @@ def choose_inhouse_machine(self, loanid, loan_class):
     return chosen['name']
 
 
+@task(bind=True)
+@add_to_history(
+    before="Identifying aws machine name to use",
+    after="Chose aws machine {retval!s}")
+def choose_aws_machine(self, loanid, loan_class):
+    logger.debug("Choosing aws machine name")
+    # We use foo-$user_shortname$N where $N is optional only if
+    # there exists another active loan with the foo-$user prefix
+    l = Loans.query.get(loanid)
+    prefix = slave_mappings.slavetype_to_awsprefix(loan_class)
+    user_shortname = l.human.ldap.split("@")[0]
+    bare_name = prefix + "-" + user_shortname
+    similar_loans = Loans.query \
+                         .filter(Loans.machine_id == Machines.id) \
+                         .filter(Machines.fqdn.like(bare_name + "%")) \
+                         .filter(~Loans.status.in_(["COMPLETE"])) \
+                         .order_by(Machines.fqdn.desc())
+    if similar_loans.count():
+        existing_aws_loan = similar_loans.first().machine.fqdn
+        shortname = existing_aws_loan.split(".")[0]
+        this_name = bare_name + str(int(shortname[len(bare_name):])) + 1
+    else:
+        this_name = bare_name
+    logger.debug("Chosen Slave Name = %s" % this_name)
+    return this_name
```

Comment:
Admittedly this logic and the proliferation of var names feels VERY messy/fragile to me. I'm open to options to clean it up now. But failing any I'll commit to cleaning it up "later"